Storage/ThirdParty/SwiftData.swift

```diff
@@ -50,10 +50,26 @@ class DeferredDBOperation<T>: Deferred<T>, Cancellable {
     fileprivate var dispatchWorkItem: DispatchWorkItem?
     private var _running = false
 
+    fileprivate let queue: DispatchQueue
+
+    fileprivate weak var connection: ConcreteSQLiteDBConnection?
+
+    init(queue: DispatchQueue) {
+        self.queue = queue
+    }
+
     func cancel() {
         objc_sync_enter(self)
         defer { objc_sync_exit(self) }
+
+        queue.suspend()
```

Comment:
`OperationQueue.current?.underlyingQueue.suspend()` should work instead of storing a variable for this

Reply:
Ah, good call. I didn't know you could do this.